arch.mk

```diff
@@ -62,13 +62,39 @@ endif
 ## ARM Cortex-A
 ifeq ($(ARCH),AARCH64)
   CROSS_COMPILE?=aarch64-none-elf-
-  CFLAGS+=-DARCH_AARCH64 -march=armv8-a
+  CFLAGS+=-DARCH_AARCH64
   OBJS+=src/boot_aarch64.o src/boot_aarch64_start.o
-  CFLAGS+=-DNO_QNX
+
+  ifeq ($(TARGET),nxp_ls1028a)
+    ARCH_FLAGS=-mcpu=cortex-a72+crypto -march=armv8-a+crypto -mtune=cortex-a72
+    CFLAGS+=$(ARCH_FLAGS) -DCORTEX_A72
+
+    CFLAGS +=-ffunction-sections -fdata-sections
+    LDFLAGS+=-Wl,--gc-sections
+
+    ifeq ($(DEBUG_UART),0)
+      CFLAGS+=-fno-builtin-printf
+    endif
+
+    SPI_TARGET=nxp
+  else
+    # By default disable ARM ASM for other targets
+    NO_ARM_ASM=1
```

Comment:
This should perhaps still be `NO_ARM_ASM?=1` so it can be overridden in other targets